### DISCUSSION TITLE
Avoiding temporarily closed roads?

### DISCUSSION THREAD

**Discussion by Miq1** [organicmaps/organicmaps#8310] [Q&A]
Is there a way to signal closed roads to the routing mechanism?

A current example is the A12 motorway in the Netherlands around the city of Utrecht in June. As the A12 is a very central connection, OM will use it as part of routes normally. Due to the constructipn works it will be no valid choice any more for some weeks.

I found no means yet to prevent the routing from using this road - is there a technique to achieve that?

**Comment by biodranik** [comment]
Is it marked in OSM as construction work?

**Reply by Miq1** [reply]
> Is it marked in OSM as construction work?

No, the construction works are temporarily only. Complete closing seems to be happening at nights and weekends, one lane shall be open during daytime. Given the usual traffic that will be like a closure anyway.

**Reply by biodranik** [reply]
Was any temporary construction work tag introduced? There are planned closure dates, and they can be specified explicitly for many cases.

**Reply by Miq1** [reply]
Where would that be done - Openstreetmap? 
Yes, see below.

Will the modifications vanish again after the times will have passed?
No, there is no mechanism to automatically clean up these attrobutes yet.

In the meantime I found an attribute option in OSM to mark the road accordingly:
```
motor_vehicle:conditional=no @ 2024 June 22-2024 July 13
```

**Comment by muralito** [comment]
Supposing is only temporarily closed, it seems it should not be tagged in OSM.

I remember that my old Garmin Nuvi has a feature called "Custom Avoidances".
Options are 1) Add avoid area => takes you to the map, ask you to define a rectangle, clicking in one corner, then the opposite corner and then confirming the area to avoid.
2) Add avoid road => takes you to the map, ask to touch the avoid road start point, then the avoid road end point, and then confirm the road to avoid.
This feature lets you avoid the road or area until you delete or disable the configuration.

Another avoidance feature is in driving mode, it lets you ask for a detour with four options: next 0.5 km on route, next 2 km on route, next 5 km on route, detour by road on route.
the first three goes to calculating a new route with the selected detour. the latest one presents a list of roads (on the route) to avoid, you select one of them and it calculates a new route that doesn't use the selected road.
This feature is not permanent. You need to use every time you need it.

**Reply by Misalf-git** [reply]
OsmAnd has an "avoid this road" feature to exclude user selections from routing.

**Reply by Miq1** [reply]
That would be the ideal solution IMHO.

**Reply by biodranik** [reply]
This button can be added when a route is built or navigation is active. A related road should be selected/highlighted to a user. Was there already a related issue? If no, let's create it.

**Reply by Ty2357364** [reply]
> This button can be added when a route is built or navigation is active. A related road should be selected/highlighted to a user. Was there already a related issue? If no, let's create it.

Did an issue get made?

**Comment by hb0nd** [comment]
Huh, i thought there was already an issue open for this, but can't find one.

I definitely agree, there are many reasons that can't be included in OSM as to why a road should be avoided, eg. you hear there was a fire/bridge collapse on the news, or there's construction work only happening for a couple of days.

**Reply by biodranik** [reply]
There should be a way for anyone in the world to report some temporary map issue. Are there any existing projects that rely on OSM data?

**Reply by muralito** [reply]
> There should be a way for anyone in the world to report some temporary map issue. Are there any existing projects that rely on OSM data?

Some issues may be temporary, but others not, like "zonas rojas" in LATAM (high risk of crime, robbery, gangs, few police control, ...), these geographic data is not and maybe will never be in OSM because of OSM policies. This is needed to e.g. do not guide a user/tourist to go via an area in which it could be risky. 

Other map applications,like Garmin Nuvi, allows you to add this kind of avoidances, downloading "zonas rojas" data from a third party provider ans installing in your device (sometimes free, sometimes propietary) 

As this kind of geographic data is more or less permanent, it also could be added as a OM feature besides the OSM data, so the user doesn't have to search outside OM. Anyway, allowing OM to install custom avoidances zones, could lead some users to begin sharing their data in this format.

**Reply by biodranik** [reply]
I think there already was a related issue about "avoid danger zones", right? Let's start collecting these polygons/their sources there, to transparently integrate them into the routing's "Avoid..." options later.

We also need a solution for temporary road works/closures, and accidents that block the road. Is there already an issue about it?

**Reply by Miq1** [reply]
I think there are two different applications here:
- mark a polygonal area, for "red zones" etc.
- mark one or more segments of a road, for construction closures or traffic jams

While the former will have somewhat longer lifetimes, the latter tends to be lasting weeks, days or even hours only.

Both may be local to a single user or of general interest. Local markings better will be held on the user's device only, while the general ones are better suited as a central map item.

**Reply by biodranik** [reply]
The idea is to share this information with other users in a privacy-friendly manner so everyone can benefit.

**Reply by Miq1** [reply]
There already is the functionality to long press on a spot on the map to pop up the details about it and set a bookmark etc. It would be handy to have that extended to also allow for entering road blocks (if the mark is on or close to a road) or put it on a list for later refinement (like defining a polygonal area, mentioning a map issue or quickly bookmark a group of these spots).

**Comment by Darin755** [comment]
In OsmAnd there is an option to avoid a road. Is there something Simular in organic maps? My workaround is to carefully add stops. It would be nice to have a button to avoid a stretch of road. Often roads get closed for  various reasons and it wouldn't make sense to create an entry on OSM for a short time.

**Comment by regenman** [comment]
Potentially interesting for this discussion: 

https://github.com/sosm/temporary-road-closures

**Comment by ORCHORDS** [comment]
There's no "avoid this road" button in Organic Maps, so the fix lives in **OpenStreetMap** — that's the data the app routes on.

If the closure has a known end date (like your A12 roadworks), add a conditional restriction to the way itself. Open the road at openstreetmap.org, hit Edit, and add:

- `access=no` (or `vehicle=no` if pedestrians can still get through)
- `vehicle:conditional=no @ (2026 Sep 01-2026 Dec 15)` — swap in the real dates

Save with a comment referencing the municipal notice so other mappers know it's legit.

Closed for the long haul? Then the OSM wiki says to only use `highway=construction` for closures of **six to nine months or more**. In that case retag the segment as `highway=construction` plus `construction=motorway` (its old class), and throw on `opening_date` if you know it.

One thing to skip: plain `construction=yes`. The wiki flags it as breaking routers that don't understand the tag.

After the edit lands, update the Netherlands map in the app — or just wait for the periodic refresh — and routing will steer around the A12 until your end date passes.
